### PR TITLE
Refactor Mulliken Partitioning to Domain

### PR DESCRIPTION
**Pull request by xdvriend**
**Short description**

This PR refactors the MullikenPartitioning functionality to the new domain functionality.

**Related issues**

[This issue in GQCC](https://github.com/GQCG-res/GQCConstraints/issues/54#event-8063738641).

**To do**

- [x] Generalised case
- [x] Restricted case
- [x] Unrestricted case
- [x] Extensive testing
- [x] refactor python bindings

### CODE REVIEW

**Comment by xdvriend** [comment]
Current status:

- All tests working (locally)
- Tests failing on Github is due to python bindings

Next step:
- Fix python bindings

File: gqcp/include/Domain/MullikenDomain/GMullikenDomain.hpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+#include "Mathematical/Representation/SquareMatrix.hpp"
+#include "QuantumChemical/SpinResolved.hpp"
+
+
+namespace GQCP {
+
+
+/**
+ * A Mulliken partitioned domain as a collection of atomic orbitals.
+ * The sites {`i'} that are present in the domain are represented by a set bit at the corresponding indices `i'.
```
Comment (dariavh) [inline gqcp/include/Domain/MullikenDomain/GMullikenDomain.hpp]:
```suggestion
 * The sites {`i`} that are present in the domain are represented by a set bit at the corresponding indices `i`.
```

File: gqcp/include/Domain/MullikenDomain/RMullikenDomain.hpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#include "Mathematical/Representation/SquareMatrix.hpp"
+#include "QuantumChemical/SpinResolved.hpp"
+
+
+namespace GQCP {
+
+
+/**
+ * A Mulliken partitioned domain as a collection of atomic orbitals.
+ * The sites {`i'} that are present in the domain are represented by a set bit at the corresponding indices `i'.
```
Comment (dariavh) [inline gqcp/include/Domain/MullikenDomain/RMullikenDomain.hpp]:
```suggestion
 * The sites {`i`} that are present in the domain are represented by a set bit at the corresponding indices `i`.
```

File: gqcp/include/Domain/MullikenDomain/UMullikenDomainComponent.hpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+#include "Mathematical/Representation/SquareMatrix.hpp"
+#include "QuantumChemical/SpinResolved.hpp"
+
+
+namespace GQCP {
+
+
+/**
+ * A Mulliken partitioned domain as a collection of atomic orbitals.
+ * The sites {`i'} that are present in the domain are represented by a set bit at the corresponding indices `i'.
```
Comment (dariavh) [inline gqcp/include/Domain/MullikenDomain/UMullikenDomainComponent.hpp]:
```suggestion
 * The sites {`i`} that are present in the domain are represented by a set bit at the corresponding indices `i`.
```

**Review by dariavh — APPROVED** [review]

**Comment by dariavh** [comment]
Apart from some small inconsistencies in the documentation regarding the typesetting of `i`, this one is good to go after the Python bindings have been tested. 👍🏻

**Pull request merged**